debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentState.java

```diff
@@ -352,8 +352,19 @@ private Headers makeHeaders(List<FieldReference> additionalHeaders, Struct origi
                 }
                 continue;
             }
-            headers.add(fieldReference.getNewFieldName(), fieldReference.getValue(originalRecordValue),
-                    fieldReference.getSchema(originalRecordValue.schema()));
+            try {
+                Schema fieldSchema = fieldReference.getSchema(originalRecordValue.schema());
+                headers.add(fieldReference.getNewFieldName(), fieldReference.getValue(originalRecordValue), fieldSchema);
+            }
+            catch (Exception e) {
+                // skip unknown field from origin record value struct
+                if (e.getMessage().startsWith("Unexpected field name:")) {
```

Comment:
@harveyyue I don't think this is correct. If the field is not supported it should be clear from the schema of the struct so if the field is not in schema it should be just ignored?